EpikCord/client/websocket_client.py

```diff
@@ -389,9 +389,39 @@ async def _interaction_create(self, data: discord_typings.InteractionCreateData)
         await self.dispatch("interaction_create", interaction)
 
     async def _channel_create(self, data: discord_typings.ChannelCreateData):
+        """Event Fired when a channel is created
+
+        :return channel: The Channel which was created
+        :type channel: EpikCord.abstract.GuildChannel
+        """
         channel = self.utils.channel_from_type(data)
         self.channels.add_to_cache(channel.id, channel)
         await self.dispatch("channel_create", channel)
+    
+    async def _channel_update(self,data:discord_typings.ChannelUpdateData):
+        """Event Fired when a channel is modified/updated
+
+        This event dispatches both the channel before the modification and the channel after the modification
+
+        :return channel_before: The Channel object before the modification
+        :type channel: EpikCord.abstract.GuildChannel
+        :return channel_before: The Channel object after the modification
+        :type channel: EpikCord.abstract.GuildChannel
+        """
+        channel_after = self.utils.channel_from_type(data)
+        channel_before = self.channels.get(channel_after.id)
+        self.channels.add_to_cache(channel_after.id, channel_after)
+        await self.dispatch("channel_update", channel_before, channel_after) 
+
+    async def _channel_delete(self, data:discord_typings.ChannelDeleteData):
+        """Event Fired when a channel is deleted
+
+        :return channel: The Channel which was deleted
+        :type channel: EpikCord.abstract.GuildChannel
+        """
+        channel = self.utils.channel_from_type(data)
+        self.channels.remove_from_cache(channel.id)
```

Comment:
To save some CPU work, what you could do is just extract the `id` of `data`, and then read it and then remove it.
So that would be
```diff
-       channel = self.utils.channel_from_type(data)
+       channel = self.channels.get(int(data["id"]))
        self.channels.remove_from_cache(int(data["id"]))
        await self.dispatch("channel_delete", channel)
```
This saves the creation and the `self.X = X`, and just reading the already existing item

Reply:
Great idea!

Reply:
implemented in [6ed5b5b](https://github.com/EpikCord/EpikCord.py/pull/208/commits/6ed5b5b0542d873f2cb8341b4a46790affdb065c)